### PR TITLE
Source files using `--login` for bash and zsh

### PR DESCRIPTION
**Pull request by oxalica**
Simplify codes and make `zsh` also source `/etc/profile`.
`/etc/profile` contains basic environment variables like `QT_IM_MODULE` on NixOS. Ignoring it will make input method unusable.

### CODE REVIEW

**Comment by marcthe12** [comment]
there is Xsession script also which performs the same logic. I suggest to do the necessary changes to that script also

**Comment by oxalica** [comment]
@marcthe12 Fixed.

**Review by davidedmundson — APPROVED** [review]
It does indeed seem simpler and appears to work the same

File: data/scripts/Xsession [hunk 1]
```diff
@@ -8,26 +8,11 @@
 # Note that the respective logout scripts are not sourced.
```
Comment (tavianator) [inline data/scripts/Xsession:8]: I think this won't be true any more

Comment (oxalica) [reply data/scripts/Xsession:8]: We `exec` at the end of script, I confirmed logout scripts have no possibility to run when we  `exec`.

Comment (tavianator) [reply data/scripts/Xsession:8]: Oh true, never mind!

**Pull request merged**